### ISSUE TITLE
Unsure where to store ERC20 Token for reward pool

### ISSUE THREAD

**Issue by nickyoung92** [animocabrands/ethereum-contracts-nft_staking#15]
I see the constructor is there to bring in the erc20 contract, but how does one provide the staking contract with the rewards pool? Do we send the erc20 token to the staking contract?

**Comment by nataouze** [comment]
Yes, when the owner calls the function `addRewardsForPeriods`, the necessary amount of ERC20 will be computed and the total will be transferred to the contract. So the owner needs to own enough ERC20 and to `approve` the staking contract before calling the function
